modules/core/src/main/java/org/apache/synapse/message/store/impl/rabbitmq/RabbitMQConsumer.java

```diff
@@ -34,255 +34,220 @@
+                    MessageContext synapseMc = store.newSynapseMc(axis2Mc);
+                    synapseMc = MessageConverter.toMessageContext(storableMessage, axis2Mc, synapseMc);
+                    updateCache(delivery, delivery.getProps().getMessageId());
+                    log.info(getId() + " Received MessageId: " + delivery.getProps().getMessageId());
+                    if (log.isDebugEnabled()) {
+                        log.debug(getId() + " Received MessageId: " + delivery.getProps().getMessageId());
+                    }
+                    return synapseMc;
+                }
+            } catch (ShutdownSignalException | IOException e) {
+                log.error(getId() + " connection error when receiving messages." + e);
+            }
+        } else {
+            cleanup();
+            setConnection(store.createConnection());
+            setChannel(store.createChannel(connection));
+        }
+        return null;
+    }
 
-	public boolean ack() {
-		boolean result = cachedMessage.ack();
-		if (result) {
-			store.dequeued();
-		}
-		return result;
-	}
+    /**
+     * Deserialize the message to store in the queue
+     *
+     * @param channel  the {@link Channel} object to acknowledge the message in deserialization failure
+     * @param delivery the message received from the broker
+     * @return a {@link StorableMessage} object
+     * @throws IOException
+     */
+    private StorableMessage deserializeMessage(Channel channel, GetResponse delivery) throws IOException {
+        StorableMessage storableMessage = null;
+        try (ByteArrayInputStream inputStream = new ByteArrayInputStream(delivery.getBody());
+             ObjectInput objectInput = new ObjectInputStream(inputStream)) {
+            storableMessage = (StorableMessage) objectInput.readObject();
+        } catch (ClassNotFoundException e) {
+            log.error(getId() + "unable to read the stored message." + e);
+            channel.basicAck(delivery.getEnvelope().getDeliveryTag(), false);
+        }
+        return storableMessage;
+    }
 
-	public boolean cleanup() {
-		if (logger.isDebugEnabled()) {
-			logger.debug(getId() + " cleaning up...");
-		}
-		boolean result = store.cleanup(connection, true);
-		if (result) {
-			connection = null;
-			return true;
-		}
-		return false;
-	}
+    /**
+     * Acknowledge the message upon successful backend invocation
+     *
+     * @return whether message is successfully acknowledge or not
+     */
+    @Override
+    public boolean ack() {
+        boolean result = cachedMessage.ack();
+        if (result) {
+            store.dequeued();
+        }
+        return result;
+    }
 
+    /**
+     * Cleanup the {@link Connection} and it's related resources
+     *
+     * @return whether cleanup successful or not
+     */
     @Override
-    public boolean isAlive() {
+    public boolean cleanup() {
+        if (connection != null) {
+            connection.abort();
```

Comment:
what does this method do? Is it closing the connection and release the network resources? Otherwise there can be a connection leak here.

Reply:
It supposes to close the connection and its channel forcefully.